src/auth/session/session.ts

```diff
@@ -1,32 +1,32 @@
+// import {Context} from '../../context';
 import {OnlineAccessInfo} from '../oauth/types';
 
 /**
  * Stores App information from logged in merchants so they can make authenticated requests to the Admin API.
  */
 class Session {
   public static cloneSession(session: Session, newId: string): Session {
-    const newSession = new Session(newId);
+    const newSession = new Session(newId, session.shop, session.state, session.isOnline);
 
-    newSession.shop = session.shop;
-    newSession.state = session.state;
     newSession.scope = session.scope;
     newSession.expires = session.expires;
-    newSession.isOnline = session.isOnline;
     newSession.accessToken = session.accessToken;
     newSession.onlineAccessInfo = session.onlineAccessInfo;
 
     return newSession;
   }
 
-  public shop: string;
-  public state: string;
-  public scope: string;
+  public scope?: string;
   public expires?: Date;
-  public isOnline?: boolean;
   public accessToken?: string;
   public onlineAccessInfo?: OnlineAccessInfo;
 
-  constructor(readonly id: string) {}
+  constructor(
+    readonly id: string,
+    public shop: string,
+    public state: string,
+    public isOnline: boolean = false,
```

Comment:
I wonder if we want the default here - I'd prefer to always have the session be explicitly defined as online / offline.

Reply:
That's a good question. We've always defaulted to offline and I don't think ever revisited that decision? I'll see what breaks if we remove it!